test/tests/functional/pbs_release_limited_res_suspend.py

```diff
@@ -929,3 +929,68 @@ def test_suspended_job_gets_calendered(self):
         self.scheduler.log_match(
             jid1 + ";Can't find start time estimate", existence=False,
             max_attempts=2)
+
+    def helper_test_preempt_release_all(self, preempt_method):
+        """
+        Helper function to test that when preempting jobs, all resources
+        are released during preemption simulation for R and C methods
+        """
+        if preempt_method == "R":
+            schedlog_msg = "Job preempted by requeuing"
+        elif preempt_method == "C":
+            schedlog_msg = "Job preempted by checkpointing"
+        else:
+            raise Exception("Unexpected value of argument preempt_method: %s"
+                            % (preempt_method))
+
+        a = {ATTR_restrict_res_to_release_on_suspend: 'mem'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        self.scheduler.set_sched_config({'preempt_order': preempt_method})
+
+        # Set 1gb mem available on the node
+        a = {ATTR_rescavail + '.ncpus': "2"}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+
+        # Submit a low priority jobs which takes up all of the ncpus
+        j1 = Job(TEST_USER)
+        j1.set_attributes({ATTR_l + '.select': '1:ncpus=2'})
```

Comment:
ATTR_restrict_res_to_release_on_suspend is set to mem in the test but the job is not requesting it, is it intentional?

Reply:
Yes it is. The idea is to create a situation where PBS will NOT release anything other than mem. So, when the low priority job requests ncpus, PBS will not release ncpus from the nodes for suspend, but it should release them for checkpoint and requeue